sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -983,6 +983,32 @@ object WindowSpecCheck extends ExprChecks {
   }
 }
 
+object CreateMapCheck extends ExprChecks {
+
+  // Spark supports all types except for Map for key and value (Map is not supported
+  // even in nested types)
+  val keyValueSig: TypeSig = (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 +
+    TypeSig.ARRAY + TypeSig.STRUCT).nested()
+
+  override def tag(meta: RapidsMeta[_, _, _]): Unit = {
+    if (meta.childExprs.length > 2 && !meta.conf.isCreateMapEnabled) {
```

Comment:
Do we want to check if all of the values are literal values, and if so allow it through, assuming there are no duplicates?

Reply:
Sure.  Could you give me some pointers on how to check for duplicates using cuDF? I assume I would have to concat the key columns first but I am not sure where to go from there.

Reply:
Never mind ... these are literal values in the expression tree ... got it